jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -61,8 +63,14 @@ public static void extract(Path source, Path destination) throws IOException {
           if (entryPath.getParent() != null) {
             Files.createDirectories(entryPath.getParent());
           }
-          try (OutputStream out = new BufferedOutputStream(Files.newOutputStream(entryPath))) {
-            ByteStreams.copy(tarArchiveInputStream, out);
+
+          String linkName = entry.getLinkName();
+          if (!Strings.isNullOrEmpty(linkName)) {
+            Files.createSymbolicLink(entryPath, Paths.get(linkName));
```

Comment:
So what happens when we then process this layer? Do we lose this symlink?

Reply:
The `.extract()` method is only used when using a local tar or Docker daemon base image. And we'll see a symlink only when there are duplicate layers. I learned that when our code reads through the symlink, it actually reads the original file. It's not possible that you read "something" out of the symlink itself.

Reply:
Ah right, so we only see symlinks to the layer tar (but we never expand out those layer tars).